test/hotspot/jtreg/compiler/whitebox/ReplaceNMethodVerifyNoRecomp.java

```diff
@@ -0,0 +1,156 @@
+
+        verifyOutput(output.getOutput());
+    }
+
+    // Matches output for function being compiled. Does not match to deoptimization outputs for that function
+    // Example: 4       compiler.whitebox.ReplaceNMethodVerifyNoRecomp$ReplaceNMethod::function (20 bytes)
+    private static String methodCompiledRegex = "^4\\s+compiler\\.whitebox\\.ReplaceNMethodVerifyNoRecomp\\$ReplaceNMethod::function\\s+\\(\\d+\\s+bytes\\)\\\n$";
+
+    public static void verifyOutput(String text) {
+        int notCompiled = text.indexOf("Should not be compiled");
+        if (notCompiled == -1) {
+            throw new RuntimeException("Output does not include required out");
+        }
+
+        int functionCompiled = text.indexOf("4       compiler.whitebox.ReplaceNMethodVerifyNoRecomp$ReplaceNMethod::function");
+        if (functionCompiled == -1) {
+            throw new RuntimeException("Function was never compiled");
+        }
+
+        // Confirm that the function was not compiled when it was not supposed to be
+        if (functionCompiled < notCompiled) {
+            throw new RuntimeException("Function was compiled before it should be");
+        }
+
+        int isCompiled = text.indexOf("Should be compiled");
+        if (isCompiled == -1) {
+            throw new RuntimeException("Output does not include required out");
+        }
+
+        // Confirm that the function was compled when it was supposed to be
+        if (functionCompiled > isCompiled) {
+            throw new RuntimeException("Function was not compiled after it should be");
+        }
+
+        // Confirm that the function never gets recompiled
+        String remainingOutput = text.substring(functionCompiled + 1);
+
+        // Confirm that the function never gets recompiled
+        boolean functionRecompiled = remainingOutput.matches(methodCompiledRegex);
+        if (functionRecompiled) {
+            throw new RuntimeException("Function was recompiled when it should not be");
+        }
+    }
+
+    private static class ReplaceNMethod {
+        private static final WhiteBox WHITE_BOX = WhiteBox.getWhiteBox();
+        public static double FUNCTION_RESULT = 0;
+
+        public static void main(String [] args) throws Exception {
+            // Get method that will be relocated
+            Method method = ReplaceNMethodVerifyNoRecomp.ReplaceNMethod.class.getMethod("function");
+            WHITE_BOX.testSetDontInlineMethod(method, true);
+
+            // Verify not initially compiled
+            System.out.println("Should not be compiled");
+            CompilerWhiteBoxTest.checkNotCompiled(method, false);
+
+            // Call function enough to compile
+            callFunction();
+
+            // Verify now compiled
+            System.out.println("Should be compiled");
+            CompilerWhiteBoxTest.checkCompiled(method, false);
+
+            // Get newly created nmethod
+            NMethod origNmethod = NMethod.get(method, false);
```

Comment:
What do you think about adding a loop around the copy/replacement of the method? I think it would make the test more convincing and may not impact it's execution time that much.